packages/effect/src/TQueue.ts

```diff
@@ -262,7 +262,7 @@ export const isFull: <A>(self: TQueue<A>) => STM.STM<boolean> = internal.isFull
  * @since 2.0.0
  * @category getters
  */
-export const isShutdown: <A>(self: TQueue<A>) => STM.STM<boolean> = internal.isShutdown
+export const isShutdown: <A>(self: TDequeue<A> | TEnqueue<A>) => STM.STM<boolean> = internal.isShutdown
```

Comment:
Can these remain as `TQueue`?

Reply:
I'll have to check, IIRC it doesn't typecheck as TQueue but I don't remember why

Reply:
actually this is better typed as `TDequeue<A> | TEnqueue<A>`, given that such functionality (together with others, updated in https://github.com/Effect-TS/effect/pull/2725/commits/bd7d59669fef016236e831b2acb758da4bc19d0d) are shared between `TEnqueue` and `TDequeue` so requiring a `TQueue` would be too restrictive